src/main/java/com/cedarsoftware/util/io/WriteOptions.java

```diff
@@ -1049,4 +1146,125 @@ static Set<Class<?>> loadNonRefs() {
         });
         return nonRefs;
     }
+
+
+    ///// ACCESSOR PULL IN ???????
+
+    public Map<String, Accessor> getDeepAccessorMap(Class<?> classToTraverse) {
+        return this.deepAccessorCache.computeIfAbsent(classToTraverse, this::buildDeepAccessors);
+    }
+
+    public Collection<Accessor> getDeepAccessors(Class<?> c) {
+        return this.getDeepAccessorMap(c).values();
+    }
+
+    public void clearCaches() {
+        deepAccessorCache.clear();
+        deepMethodCache.clear();
+        deepFieldCache.clear();
+        methodCache.clear();
+    }
+
+    public void clearUnfilteredAccessorCache() {
+        methodCache.clear();
+    }
+
+    private Map<String, Accessor> buildDeepAccessors(Class<?> c) {
+        final Map<String, Field> deepDeclaredFields = deepFieldCache.computeIfAbsent(c, this::buildDeepFieldMap);
+        final Map<String, Method> possibleMethods = deepMethodCache.computeIfAbsent(c, this::buildDeepMethods);
+        final Map<String, Accessor> accessorMap = new LinkedHashMap<>();
+
+        for (Map.Entry<String, Field> entry : deepDeclaredFields.entrySet()) {
+
+            final Field field = entry.getValue();
+
+            String fieldName = entry.getKey();
+            String key = accessorMap.containsKey(fieldName) ? field.getDeclaringClass().getSimpleName() + '.' + fieldName : fieldName;
+
+            Optional<Accessor> accessor = this.accessorFactories.stream()
+                    .map(factory -> {
+                        try {
+                            return factory.createAccessor(field, this.nonStandardMappings, possibleMethods, key);
+                        } catch (Throwable t) {
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .findFirst();
+
+
+
+            //  If no accessor was found, let's use the default tried and true field accessor.
+            accessorMap.put(key, accessor.orElseGet(() -> {
+                try {
+                    return new Accessor(field, key);
+                } catch (ThreadDeath td) {
+                    throw td;
+                } catch (Throwable t) {
+                    return null;
+                }
+            }));
+        }
+
+        return accessorMap;
+    }
+
+    /**
+     * Builds a list of methods with zero parameter methods taking precedence over overrides
+     * for a given single level class.
+     *
```

Comment:
If classes have overridden method names, a String key of the method + "(" + param0.typeName +")" AND method + "(" + param0.typeName + ", " + param1.typeName + ") AND ... will generate unique methods that will each have their own entry in the map.  This is a deterministic way to no lose any methods to overwriting in key of the Map.  Now, if you do it this way, would it be helpful / make it work better?